edit-post/store/effects.js

```diff
@@ -90,6 +89,31 @@ const effects = {
 		const message = action.mode === 'visual' ? __( 'Visual editor selected' ) : __( 'Code editor selected' );
 		speak( message, 'assertive' );
 	},
+	INIT( _, store ) {
+		// Select the block settings tab when the selected block changes
+		subscribe( onChangeListener(
+			() => select( 'core/editor' ).getBlockSelectionStart(),
+			( selectionStart ) => {
+				const isSidebarOpened = select( 'core/edit-post' ).isEditorSidebarOpened();
```

Comment:
This function could exit early when the sidebar is not opened. This would simplify logic a bit:
```js
if ( ! isSidebarOpened ) {
    return;
}
store.dispatch(
    openGeneralSidebar( `edit-post/${ selectionStart ? 'block' : 'document' }` )
);
```